lib/pure/httpclient.nim

```diff
@@ -713,10 +713,10 @@ proc downloadFile*(url: string, outputFilename: string,
 proc generateHeaders(requestUrl: Uri, httpMethod: string,
                      headers: HttpHeaders, body: string, proxy: Proxy): string =
   # GET
-  result = httpMethod.toUpper()
+  result = httpMethod.toUpperAscii()
   result.add ' '
 
-  if proxy.isNil:
+  if proxy.isNil or (not proxy.isNil and requestUrl.scheme == "https"):
     # /path?query
```

Comment:
I want to double check this. As it is right now:

* A request to an https site via a proxy will send a URL with the path and query to the proxy
* A request to an http site via a proxy will just send the hostname and port.

Why the discrepancy?

Reply:
Following are all the cases to covered ```generateHeaders()```:-
- http direct - talking to server directly
  - Send path: ```if proxy.isNil```
- https direct - talking to server over encrypted socket directly
  - Send path: ```if proxy.isNil```
- http through proxy - talking to proxy who needs full URL + path
  - Send full URL: ```else + requestUrl.scheme != "https"```
- https through proxy
  - First send host:port for CONNECT tunnel: ```else + requestUrl.scheme == "https"```
    - Also making sure we send only proxy host:port for requestUrl in newConnection() line 1097
  - Next send path: ```not proxy.isNil and requestUrl.scheme == "https"``` since talking through tunnel directly with server

Reply:
I'm still rather confused by this, but I will simply trust you and merge this :)